### PR TITLE
Analytics updates

### PR DESCRIPTION
**Pull request by chrisdugne**
many fixes for analytics tracking.

~~✅ requires https://github.com/CoorpAcademy/components/pull/1565~~

### CODE REVIEW

**Comment by chrisdugne** [comment]
[Retours GTM](https://trello.com/c/4poy0K66/1279-retours-gtm)

File: src/components/correction.js [hunk 1]
```diff
@@ -284,6 +284,15 @@ class Correction extends React.PureComponent<Props> {
 
     const cards = this.createCards();
 
+    let analyticsID;
+    if (canGoNext) {
+      analyticsID = 'button-next-question';
+    } else if (offeringExtraLife) {
+      analyticsID = 'button-refuse-extralife';
+    } else {
+      analyticsID = 'button-game-over';
```
Comment (VincentCATILLON) [inline src/components/correction.js]: You're breaking backward compatibility for the tracking plan. Tell to PO your changes

File: src/components/level-end.js [hunk 2]
```diff
@@ -280,6 +280,7 @@ class LevelEnd extends React.PureComponent<Props> {
             <ButtonSticky
               onPress={this.handleButtonPress}
               testID={`button-${isSuccess ? 'next' : 'retry'}-level`}
+              analyticsID={`button-level-end-${isSuccess ? 'next' : 'retry'}-level`}
```
Comment (VincentCATILLON) [inline src/components/level-end.js]: 👍

File: src/containers/cards-swipable.js [hunk 3]
```diff
@@ -60,6 +67,11 @@ class CardsSwipable extends React.PureComponent<Props, State> {
 
   componentDidMount = () => {
     this.hint.start();
+
+    const cardOnTop = this.props.items[0];
```
Comment (VincentCATILLON) [inline src/containers/cards-swipable.js]: cardOnTop can be undefined

File: src/containers/cards-swipable.js [hunk 4]
```diff
@@ -60,6 +67,11 @@ class CardsSwipable extends React.PureComponent<Props, State> {
...
+    if (cardOnTop.resource) {
```
Comment (VincentCATILLON) [inline src/containers/cards-swipable.js]: And so, it should be checked there

File: src/containers/cards-swipable.js [hunk 5]
```diff
@@ -96,6 +108,11 @@ class CardsSwipable extends React.PureComponent<Props, State> {
 
     analytics && analytics.logEvent(ANALYTICS_EVENT_TYPE.SWIPE, data);
 
+    const cardOnTop = items[(cardIndexSwiped + 1) % items.length];
+    if (cardOnTop.resource) {
```
Comment (VincentCATILLON) [inline src/containers/cards-swipable.js]: Same as above

File: src/redux/actions/permissions.js [hunk 6]
```diff
@@ -61,6 +61,11 @@ const _requestPermission = (type: PermissionType, onDeny?: () => void) => async
     dispatch(change(type, status));
   }
 
+  services.Analytics.logEvent(ANALYTICS_EVENT_TYPE.PERMISSION, {
+    status,
+    type
+  });
```
Comment (VincentCATILLON) [inline src/redux/actions/permissions.js]: 👍

File: src/redux/middlewares/update-card-on-progression-update.js [hunk 7]
```diff
@@ -19,7 +19,7 @@ Middleware<State, StoreAction<Action>, Dispatch<StoreAction<Action>>> => ({
   dispatch,
   getState
 }) => next => action => {
-  if (action.type === UI_PROGRESSION_ACTION_TYPES.PROGRESSION_UPDATED) {
+  if (action.type === PROGRESSION_UPDATED_ON_MOVE || action.type === PROGRESSION_UPDATED_ON_NODE) {
```
Comment (VincentCATILLON) [inline src/redux/middlewares/update-card-on-progression-update.js]: not requested: [PROGRESSION_UPDATE_ON_MOVE, PROGRESSION_UPDATED_ON_NODE].includes(action.type) is easier to read IMO

File: src/components/correction.js (hunk 1 above)
Comment (chrisdugne) [reply src/components/correction.js]: @MelkMerle is aware, @marcmdm 
```js
    let analyticsID;
    if (canGoNext) {
      analyticsID = 'button-next-question';
    } else if (offeringExtraLife) {
      analyticsID = 'button-refuse-extralife';
    } else {
      analyticsID = 'button-game-over';
    }
```

File: src/redux/middlewares/update-card-on-progression-update.js (hunk 7 above)
Comment (chrisdugne) [reply src/redux/middlewares/update-card-on-progression-update.js]: done

**Pull request merged**